tasks/main.yml

```diff
@@ -0,0 +1,127 @@
+---
+- name: Gather needed facts
+  setup:
+    gather_subset:
+      - '!all'
+      - '!min'
+      - distribution_major_version
+  when: ansible_facts.distribution_major_version is not defined
+
+- name: System check
+  fail:
+    msg: >-
+      Only Red Hat Enterprise Linux >= 8.1 is supported;
+      System: {{ ansible_facts.os_family }}
+      Version: {{ ansible_facts.distribution_version }}
+  when: (ansible_facts.os_family != "RedHat") or
```

Comment:
I guess this should also support CentOS, how about Fedora? Should be like this:
```
  when: (ansible_distribution not in ['CentOS', 'RedHat']) or
    (ansible_distribution in ['CentOS', 'RedHat'] and
    ansible_distribution_version is version("8.1", "<"))
```

Reply:
I'll add there fedora as well.

Reply:
It seems that {{ ansible_facts.os_family }} is "RedHat" for Fedora and CentOS. I guess no change is needed then.

Reply:
The point is to compare RHEL version (8.1 etc.) for RHEL and CentOS, and for Fedora compare its version (38 etc, which is not applicable here but make the definition more explicit).
Also, I understand that it makes sense to compare RHEL version in this case, but for specific functionalities, can the role instead check for the version of the package?

Reply:
It's hard to check for the version of the package because of lots of feature backporting in RHEL downstream. I consider that Fedora has all the features unconditionally. It's very hard to find fedora version that doesn't.

Reply:
IMO doing `ansible_facts.os_family == "RedHat" and ansible_facts.distribution_version is version("8.1", "<")` is fine for Fedora also - yes, the versions of Fedora and EL are not compatible, but version comparisons like this generally work because comparisons like EL `version("9", "<")` will exclude all Fedora versions (which is desired) and comparisons like EL `version("9", ">=")` will include all Fedora versions (which is desired).

Then, if there is some version incompatibility with Fedora in the future, we can go back and change the comparisons to include EL versions but exclude certain Fedora versions (or vice versa).